src/Psalm/Internal/Stubs/CoreGenericFunctions.phpstub

```diff
@@ -481,6 +481,13 @@ function strtolower(string $str) : string {}
  */
 function strtoupper(string $str) : string {}
 
+/**
+ * @psalm-pure
+ *
+ * @psalm-return positive-int|0|false
```

Comment:
actual point of the pr is to catch callers which falsly assume the return could be negative